code/obj/ToSplit/barber_shop.dm

```diff
@@ -437,3 +436,555 @@
+		return 0 // Who's cutting whose hair, now?
+
+	if(user.zone_sel.selecting != "head")
+		boutput(user, "You poke [M] with your [src]. You'll need to aim for [M]'s head to cut their hair. Or change your grip if you want to cut anything else.")
+		return 0
+
+	var/mob/living/carbon/human/H = M
+	if(ishuman(M) && ((H.head && H.head.c_flags & COVERSEYES) || (H.wear_mask && H.wear_mask.c_flags & COVERSEYES) || (H.glasses && H.glasses.c_flags & COVERSEYES)))
+		// you can't stab someone in the eyes wearing a mask!
+		boutput(user, "<span class='notice'>You're going to need to remove that mask/helmet/glasses first.</span>")
+		return 0
+
+	if(M.bioHolder.mobAppearance.customization_first == "None")
+		boutput(user, "<span class='alert'>There is nothing to cut!</span>")
+		return 0
+
+	if(!mutant_barber_fluff(M, user, "haircut"))
+		return 0
+
+	var/new_style = input(user, "Please select style", "Style")  as null|anything in customization_styles + customization_styles_gimmick
+
+	if (new_style)
+		if(M.bioHolder.mobAppearance.customization_first == "Balding" && new_style != "None")
+			boutput(user, "<span class='alert'>Not enough hair!</span>")
+			return 0
+
+	if(!new_style)
+		return 0
+
+	var/our_barbery_conditions = get_barbery_conditions(M, user)
+	var/degree_of_success = 0 // 0 - 3, 0 being failure, 3 being catastrophic hair success
+	if(prob(our_barbery_conditions))
+		degree_of_success = 3
+	else // oh no we fucked up!
+		degree_of_success = (our_barbery_conditions % 3) - 1
+	//and then just jam all the vars into the action bar and let it handle the rest!
+	actions.start(new/datum/action/bar/haircut(M, user, degree_of_success, new_style), user)
+	return 1
+
+/proc/get_barbery_conditions(mob/living/carbon/human/M as mob, mob/living/carbon/human/user as mob)
+	var/barbery_conditions = 0
+	// let's see how ideal the haircutting conditions are
+	if(M.stat || issilicon(user))
+		barbery_conditions = 100
+	else
+		if(istype(M.buckled, /obj/stool/chair/comfy/barber_chair))
+			barbery_conditions += 30
+
+		if(get_area(M) == /area/station/crew_quarters/barber_shop)
+			if(get_area(M) == get_area(user))
+				barbery_conditions += 30
+			else	// you should ideally be in the same room as whoever's hair you're cutting
+				barbery_conditions += 5
+
+		if(M.jitteriness)
+			barbery_conditions -= 20
+
+		if(ishuman(user))
+			if(istype(user.w_uniform, /obj/item/clothing/under/misc/barber))
+				barbery_conditions += 30
+			if(user.mind.assigned_role == "Barber") // 60% chance just for being you, 90 if you're wearing pants
+				barbery_conditions += 30
+				barbery_conditions *= 2
+			else if(M == user)
+				barbery_conditions *= 0.5
+			if(user.bioHolder.HasEffect("clumsy"))
+				barbery_conditions *= 0.3
+			if(user.jitteriness && !M.jitteriness) // your jitteriness kind of... syncs up
+				barbery_conditions -= 20
+			if(M == user)
+				barbery_conditions *= 0.5
+
+	barbery_conditions = clamp(barbery_conditions, 1, 100)
+	return (barbery_conditions)
```

Comment:
`(` `)` is redundant here.

Reply:
Forgot I left it like that, since that proc used to be part of the haircut proc.